app/services/encryption/encryptors/session_encryptor.rb

```diff
@@ -0,0 +1,26 @@
+module Encryption
```

Comment:
There is definitely much I don't understand here, but it looks like what you would want to cache here is `UserAccessKeyEncryptor.new(user_access_key)` instead of `@user_access_key_scrypt_hash`. What is the rationale for caching `@user_access_key_scrypt_hash` but returning a new `UserAccessKey` and creating a new `UserAccessKeyEncryptor` each time?